File: algo_2021/w2.-two-pointer/problems/148.sort_colors/148_sort_colors__two_pass_1.java
public class Solution {
    /**
     * @param nums: A list of integer which is 0, 1 or 2 
     * @return: nothing
     */
    public void sortColors(int[] nums) {
        // write your code here
        if (nums == null || nums.length == 0) {
            return;
        }
        
        int[] colorCnts = new int[3];
        for (int i = 0; i < nums.length; i++) {
            colorCnts[nums[i]]+=1;
        }
        
        int index = 0;
        for (int i = 0; i < colorCnts.length; i++) {
            int cnt = colorCnts[i];
            while(cnt > 0) {
                nums[index] = i;
                cnt--;
                index++;
            }
        }
    }
}

File: algo_2021/w2.-two-pointer/problems/148.sort_colors/148_sort_colors__two_pass_1.py
class Solution:
    """
    @param nums: A list of integer which is 0, 1 or 2 
    @return: nothing
    """
    def sortColors(self, nums):
        if not nums:
        return 
        
        # init cnts array 
        cnts = [0 for _ in range(3)]
        for num in nums:
           cnts[num]+= 1
        
        k = 0
        # traverse cnts array and assign new value to nums 
        for i in range(len(cnts)):
            cnt = cnts[i]
            while cnt:
                nums[k] = i
                cnt-=1
                k+=1
        return nums  